connect/resource/usage_automation.py

```diff
@@ -0,0 +1,172 @@
+            row = str(index + 2)
+            sheet['A' + row] = record.record_id
+            sheet['B' + row] = record.item_search_criteria
+            sheet['C' + row] = record.item_search_value
+            sheet['D' + row] = record.quantity
+            sheet['E' + row] = record.start_time_utc
+            sheet['F' + row] = record.end_time_utc
+            sheet['G' + row] = record.asset_search_criteria
+            sheet['H' + row] = record.asset_search_value
+        return book
+
+    def upload_spreadsheet(self, usage_file, spreadsheet):
+        # type: (File, openpyxl.Workbook) -> None
+
+        # Generate spreadsheet file
+        with NamedTemporaryFile() as tmp:
+            spreadsheet.save(tmp)
+            tmp.seek(0)
+            file_contents = tmp.read()
+
+        # Setup request
+        url = self.urljoin(self.url, usage_file.id, 'upload/')
+        headers = self.api.headers
+        headers['Accept'] = 'application/json'
+        del headers['Content-Type']  # This must NOT be set for multipart post requests
+        multipart = {'usage_file': ('usage_file.xlsx', file_contents)}
+        logger.info('HTTP Request: {} - {} - {}'.format(url, headers, multipart))
+
+        # Post request
+        try:
+            response = requests.post(url, headers=headers, files=multipart)
+        except requests.RequestException as ex:
+            raise FileCreationError('Error uploading file: {}'.format(ex))
+        logger.info('HTTP Code: {}'.format(response.status_code))
+        if response.status_code != 201:
+            msg = 'Unexpected server response, returned code {}'.format(response.status_code)
+            logger.error('{} -- Raw response: {}'.format(msg, response.content))
+            raise FileCreationError(msg)
+
+    def upload_usage_records(self, usage_file, usage_records):
+        # type: (File, List[FileUsageRecord]) -> None
+        # TODO: Using xslx mechanism till usage records json api is available
+        book = self.create_spreadsheet(usage_records)
+        self.upload_spreadsheet(usage_file, book)
+
+    def get_usage_template(self, product):
+        # type: (Product) -> str
+        location = self._get_usage_template_download_location(product.id)
+        contents = self._retrieve_usage_template(location) if location else None
+        if not location or contents is None:
+            msg = 'Error obtaining template usage file from `{}`'.format(location)
+            logger.error(msg)
+            raise FileRetrievalError(msg)
+        return contents
+
+    def submit_usage(self, usage_file, usage_records):
+        # type: (File, List[FileUsageRecord]) -> File
+        usage_file = self.create_usage_file(usage_file)
+        self.upload_usage_records(usage_file, usage_records)
+        return usage_file
+
+    def _get_usage_template_download_location(self, product_id):
+        # type: (str) -> str
+        try:
+            response = self.api.get(self.urljoin(self.config.api_url,
```

Comment:
it worth to use path= instead of url= for api get, 
all these urljoin's looks not really good

should be like:
response = self.api.get(path= '/usage/products/')
response = self.api.get(path=f '/usage/products/{product_id}')

Reply:
Is the path argument officially supported by requests? Could not find it in the documentation. I can add it to our API client though.

Reply:
Probably - good moment to invent that, for both requests and usages

Reply:
Done.